functorch/functorch/_src/partitioners.py

```diff
@@ -334,6 +333,8 @@ def ban_recomputation(node):
                 return True
             if node.target == operator.getitem:
                 return False
+            if compiler == "inductor" and node.dist_from_bw > 4:
```

Comment:
how does this help?

Reply:
It's just a heuristic to prevent rematerializing too much. I don't think it really makes that much sense tbh, but seems to help empirically. I'll add a note that it's a hack.

Reply:
nvm I'll fix it some point later :P